cli/cli_test.go

```diff
@@ -353,3 +353,71 @@ func Example_max_results() {
 	// quit
 	// node drained and shutdown: ok
 }
+
+// Example_usage is a basic test to make sure the fragile
+// help template does not break.
+// A few oddities:
+// * specifying nothing (basic usage) creates issues with
+//   with the testing flags (eg: -t.run).
+// * the flag list picks up the testing flags which are not
+//   there in a real binary.
+func Example_usage() {
+	// Don't use cliTest as we want no other flags set.
+	osExit = func(int) {}
+	osStderr = os.Stdout
+	if err := Run([]string{"help"}); err != nil {
+		fmt.Println(err)
+	}
+
+	// Output:
+	// Usage:
+	//   cockroach [command]
+	//
+	// Available Commands:
+	//   init        init new Cockroach cluster
+	//   start       start a node by joining the gossip network
+	//   cert        create ca, node, and client certs
+	//   exterminate destroy all data held by the node
+	//   quit        drain and shutdown node
+	//
+	//   log         make log files human-readable
+	//
+	//   sql         open a sql shell
+	//   kv          get, put, conditional put, increment, delete, scan, and reverse scan key/value pairs
+	//   user        get, set, list and remove users
+	//   range       list, split and merge ranges
+	//   zone        get, set, list and remove zones
+	//
+	//   version     output version information
+	//   help        Help about any command
+	//
+	// Flags:
+	//       --alsologtostderr=false: log to standard error as well as files
+	//       --color="auto": colorize standard error output according to severity
+	//   -h, --help[=false]: help for cockroach
+	//       --log-backtrace-at=:0: when logging hits line file:N, emit a stack trace
+	//       --log-dir="": if non-empty, write log files in this directory
+	//       --logtostderr=true: log to standard error instead of files
+	//       --test.bench="": regular expression to select benchmarks to run
+	//       --test.benchmem=false: print memory allocations for benchmarks
+	//       --test.benchtime=1s: approximate run time for each benchmark
+	//       --test.blockprofile="": write a goroutine blocking profile to the named file after execution
+	//       --test.blockprofilerate=1: if >= 0, calls runtime.SetBlockProfileRate()
+	//       --test.count=1: run tests and benchmarks `n` times
+	//       --test.coverprofile="": write a coverage profile to the named file after execution
+	//       --test.cpu="": comma-separated list of number of CPUs to use for each test
+	//       --test.cpuprofile="": write a cpu profile to the named file during execution
+	//       --test.memprofile="": write a memory profile to the named file after execution
+	//       --test.memprofilerate=0: if >=0, sets runtime.MemProfileRate
+	//       --test.outputdir="": directory in which to write profiles
+	//       --test.parallel=8: maximum test parallelism
```

Comment:
ah. crap. I guess I'll need to turn it into a plain test (as opposed to an example). I was afraid of this. On the plus side, it means I can filter out the test flags.